### PR TITLE
MAID-2498: test encryption

### PR DESCRIPTION
**Pull request by povilasb**
* Misc encryption tests.
* Fixed UDP echo address request bug on Mac OS X.

### CODE REVIEW

File: src/mc.rs [hunk 1]
```diff
@@ -431,4 +434,73 @@ mod tests {
...
+                .map(|(client_opt, _incoming)| {
+                    client_opt.expect("Failed to receive client connection")
+                })
+                .map(|(client_stream, _addr)| {
+                    length_delimited::Builder::new().new_framed(client_stream)
+                })
+                .and_then(|client_stream: Framed<TcpStream>| {
+                    client_stream.into_future()
+                        .map(|(request_opt, _stream)| {
+                            request_opt.expect("Failed to data from client connection")
```
Comment (canndrew) [inline src/mc.rs]: missing word ;)

File: src/mc.rs [hunk 2]
```diff
@@ -431,4 +434,73 @@ mod tests {
...
+
+            let server_info = PeerInfo::with_rand_key(server_addr);
+            let query =
+                query_public_addr(Protocol::Tcp, &addr!("0.0.0.0:0"), &server_info, &handle)
+                    .then(|_| Ok(()));
+
+            let make_query = incoming.into_future()
+                .map_err(|e| panic!(e))
+                .map(|(client_opt, _incoming)| {
+                    client_opt.expect("Failed to receive client connection")
```
Comment (canndrew) [inline src/mc.rs]: Generally in our tests we should use `unwrap!(..., "this is a panic message")` rather than `expect`.

File: src/mc.rs (hunk 1 above)
Comment (povilasb) [reply src/mc.rs]: :D

File: src/mc.rs (hunk 2 above)
Comment (povilasb) [reply src/mc.rs]: right, I'll fix it ;)

**Comment by povilasb** [comment]
@canndrew I amended the PR with mentioned fixes ;)

File: src/mc.rs [hunk 3]
```diff
@@ -431,4 +434,73 @@ mod tests {
...
+                .map(|(client_opt, _incoming)| {
+                    unwrap!(client_opt, "Failed to receive client connection")
+                })
+                .map(|(client_stream, _addr)| {
+                    length_delimited::Builder::new().new_framed(client_stream)
+                })
+                .and_then(|client_stream: Framed<TcpStream>| {
+                    client_stream.into_future()
+                        .map(|(request_opt, _stream)| {
+                            request_opt.expect("Failed to receive data from client connection")
```
Comment (canndrew) [inline src/mc.rs]: There's still an expect here :o

**Review by canndrew — APPROVED** [review]

**Pull request merged**